### PR TITLE
[WIP] Make session cookie HttpOnly

### PR DESCRIPTION
**Pull request by reggieriser**
## Description

This PR changes the session cookie so that it has HttpOnly set to true (i.e., it can only be accessed through HTTP, not JavaScript).  We did previously read the cookie in JavaScript, so there is some refactoring included in this PR that instead uses the status of the `logged_in` and persists it to our Redux store.

## Reviewer Notes

This seems to work OK from my testing, but marking it as `WIP` for the following reasons:

- A lot of client tests are failing and it's not clear to me why yet, but may have something to do with the additional data we load at login.
- We have sort of a "chicken and egg" problem in that we don't know if someone is logged in without calling the `logged_in` endpoint.  But if they're not logged in, that endpoint responds with a `401` code because it's a protected endpoint.  If having a `401` is undesirable, we'll need to do some further refactoring.

## Setup

Run the application as usual and ensure you can log in and log out with the normal experience.  Look at the session cookie and ensure it's set to `httpOnly`.  Look at the network tab in the browser dev tools and note the status of the `logged_in` endpoint.  While logged in, delete your session cookie and note if the app properly redirects you to the login page.

## Code Review Verification Steps

* [ ] This works in [Supported Browsers](./docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/164585573) for this change

### CODE REVIEW

File: src/shared/Swagger/api.js [hunk 1]
```diff
@@ -16,11 +17,19 @@ export const requestInterceptor = req => {
   return req;
 };
 
+export const responseInterceptor = res => {
+  if (res.status === 401) {
+    setLoggedOut();
+  }
+  return res;
+};
```
Comment (chrisgilmerproj) [inline src/shared/Swagger/api.js]: Yes! 👍

**Comment by chrisgilmerproj** [comment]
The client tests keep yelling because of this error:

```
TypeError: Cannot read property 'success' of undefined
```

Here's all the places where I see failing tests based on this:

```
gr GET_LOGGED_IN_USER.success src
src/scenes/Moves/ducks.js:    case GET_LOGGED_IN_USER.success:
src/scenes/Moves/Ppm/ducks.js:    case GET_LOGGED_IN_USER.success:
src/scenes/Moves/Ppm/ducks.test.js:    it('Should handle GET_LOGGED_IN_USER.success', () => {
src/scenes/Moves/Hhg/ducks.js:    case GET_LOGGED_IN_USER.success:
src/scenes/Moves/ducks.test.js:    it('Should handle GET_LOGGED_IN_USER.success', () => {
src/scenes/Moves/ducks.test.js:    it('Should handle GET_LOGGED_IN_USER.success with empty payload', () => {
src/scenes/ServiceMembers/ducks.js:    case GET_LOGGED_IN_USER.success:
src/scenes/Orders/ducks.js:    case GET_LOGGED_IN_USER.success:
src/scenes/Orders/ducks.test.js:    it('Should handle GET_LOGGED_IN_USER.success', () => {
src/shared/UI/ducks.js:    case GET_LOGGED_IN_USER.success:
src/shared/Data/users.js:    case GET_LOGGED_IN_USER.success:
```

**Comment by chrisgilmerproj** [comment]
I take it back, looks like they all fail here: https://github.com/transcom/mymove/blob/master/src/shared/UI/ducks.js#L15

Maybe I don't know the answer here but we ought to be fetching those on the redux logged in state instead of looking for the `GET_LOGGED_IN_USER.success` signal?  Or ensure that `GET_LOGGED_IN_USER` exists before we get to this part of the code?

**Comment by reggieriser** [comment]
Yes, that's what I saw too with the client test failures.  Still trying to understand the root cause.

@chrisgilmerproj, how do you feel about the 401 issue that I mentioned in the reviewer notes?  Is that a concern?

**Comment by chrisgilmerproj** [comment]
> @chrisgilmerproj, how do you feel about the 401 issue that I mentioned in the reviewer notes? Is that a concern?

I might suggest that we don't check 401 on that one API call.  Not sure how hard that is.

**Comment by reggieriser** [comment]
> I might suggest that we don't check 401 on that one API call. Not sure how hard that is.

To clarify, the user doesn't see anything different in the browser UI because of the 401 status.  This is only visible if you're looking at the API call (in the dev tools or whatever) where you'll see the red font and the 401 being returned by `logged_in`.  Since this would come up in normal usage for everyone logging in and logging out, I wasn't sure what implication that may have for any logs/alerts/etc. on the infra side.

**Comment by chrisgilmerproj** [comment]
In that case we're probably fine.  I don't think we'd have to worry.

**Comment by chrisgilmerproj** [comment]
@reggieriser @jim @chrisrcoles - I wanted to ping you all on the progress on this. I'm feeling some pressure to get this done to close out my session security epic.  This is really the last piece and I definitely need your help because I'm pretty sure I can't finish this by myself. Can you help me get an idea of when this will land?

**Comment by reggieriser** [comment]
> @reggieriser @jim @chrisrcoles - I wanted to ping you all on the progress on this.

I believe that @chrisrcoles had found a way forward the last time we spoke.  Chris, do you have an update?

**Comment by sarboc** [comment]
@chrisgilmerproj I'll be picking this up and running with it from here.

**Comment by sarboc** [comment]
Replacing with https://github.com/transcom/mymove/pull/2127 so we can start new convo threads.

**Pull request closed without merging**